### PR TITLE
Use group permissions for Release objects.

### PR DESCRIPTION
**Pull request by cverna**
This commit add support to check the permissions of
a user when trying to add, change or delete a release
We also add a scripts to add the correct permission to
the releng-team group

Signed-off-by: Clement Verna <cverna@tutanota.com>

### CODE REVIEW

**Comment by cverna** [comment]
@subpop and @abompard I ll appreciate a review of this, I am not sure that adding a separate scripts to add the permission to a group is the good way to go.
The idea is for this script to be run by Openshift when we deploy the application

File: scripts/add_group_permissions.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import django
+
+django.setup()
+
+from django.contrib.auth.models import Group, ContentType  # NOQA
+
+# Add the permission to add, change or delete a release to the releng-team group
+release_obj = ContentType.objects.get(app_label="releases")
+releases_permissions = [permission for permission in release_obj.permission_set.all()]
```
Comment (abompard) [inline scripts/add_group_permissions.py]: ...or you could just write `list(release_obj.permission_set.all())` ;-)

File: scripts/add_group_permissions.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+group, created = Group.objects.get_or_create(name="releng-team")
```
Comment (abompard) [inline scripts/add_group_permissions.py]: Ideally the group name would be taken from the command line.

File: scripts/add_group_permissions.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# This script is used to add permission to the releng-team group.
+# The permissions are checked when trying to add (POST), change (PUT or PATCH) or delete (DELETE)
+# an object.
+
+import django
+
+django.setup()
```
Comment (abompard) [inline scripts/add_group_permissions.py]: You can also use Django's script framework to write this: https://docs.djangoproject.com/en/1.11/howto/custom-management-commands/. This way you don't have to bother about settings the environment variable to get the settings, calling `setup()`, etc.

**Review by abompard — COMMENTED** [review]
When and by who is this supposed to be called? By a `.s2i/bin/run` script? In the ansible playbook as part of deployment?

File: scripts/add_group_permissions.py (hunk 3 above)
Comment (cverna) [reply scripts/add_group_permissions.py]: Nice :+1:

**Comment by cverna** [comment]
The plan was to make openshift execute this command between each pod deployment, just like we do for database migrations [0]. 


[0] - https://infrastructure.fedoraproject.org/cgit/ansible.git/tree/roles/openshift-apps/fpdc/files/deploymentconfig.yml#n21

File: scripts/add_group_permissions.py (hunk 3 above)
Comment (cverna) [reply scripts/add_group_permissions.py]: Updated

File: fpdc/releases/management/commands/add_permissions.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    def handle(self, *args, **options):
+        fas_group = options["fas_group"]
+        release_obj = ContentType.objects.get(app_label="releases")
+        releases_permissions = list(release_obj.permission_set.all())
+        group, created = Group.objects.get_or_create(name=fas_group)
+        group.permissions.set(releases_permissions)
+
+        self.stdout.write(
+            self.style.SUCCESS(f"{fas_group} Succesfully added to Releases permissions")
```
Comment (abompard) [inline fpdc/releases/management/commands/add_permissions.py]: Woah I did not know about the f-strings! That's awesome :-)
(however I think "Successfully" should be lowercase here, but that's very minor)

File: fpdc/releases/tests/test_views.py [hunk 5]
```diff
@@ -22,6 +22,11 @@ class ReleaseViewTests(APITestCase):
     @classmethod
     def setUpTestData(cls):
         cls.user = mixer.blend(User)
+        releases = ContentType.objects.get(app_label="releases")
+        releases_permissions = [permission for permission in releases.permission_set.all()]
```
Comment (abompard) [inline fpdc/releases/tests/test_views.py]: Same thing with the `list()` construct here.

**Review by abompard — COMMENTED** [review]
Looks good to me! Please fix the minor things and it's ready to go!
I'm not validating the review just now because otherwise mergify will merge it but I think it's good.

File: fpdc/releases/management/commands/add_permissions.py (hunk 4 above)
Comment (cverna) [reply fpdc/releases/management/commands/add_permissions.py]: To use without moderation :smile:

**Comment by cverna** [comment]
Thanks for the review !

File: fpdc/releases/management/commands/add_permissions.py (hunk 4 above)
Comment (abompard) [reply fpdc/releases/management/commands/add_permissions.py]: Well only if you can afford to target Python ≥ 3.6 ;-)

**Pull request merged**